packages/duckdb-wasm/src/parallel/async_connection.ts

```diff
@@ -135,17 +135,21 @@ export class AsyncResultStreamIterator implements AsyncIterable<Uint8Array> {
         if (this._depleted) {
             return { done: true, value: null };
         }
-        let buffer: Uint8Array;
+        let buffer: Uint8Array | null = null;
         if (this._inFlight != null) {
             buffer = await this._inFlight;
             this._inFlight = null;
-        } else {
+        }
+
+        while (buffer == null) {
             buffer = await this.db.fetchQueryResults(this.conn);
         }
+
         this._depleted = buffer.length == 0;
         if (!this._depleted) {
             this._inFlight = this.db.fetchQueryResults(this.conn);
```

Comment:
Should also this be a while? Unsure.

Reply:
This `fetchQueryResults` pro-actively request the engine to fetch the next chunk. Since we have the loop below we don't need to loop here.

Reply:
Thanks